kafka-rest/src/test/java/io/confluent/kafkarest/v2/MockConsumer.java

```diff
@@ -62,7 +62,7 @@ public synchronized Map<TopicPartition, OffsetAndTimestamp> offsetsForTimes(
         Map<TopicPartition, OffsetAndTimestamp> result = new HashMap<>();
         for (Entry<TopicPartition, Long> entry : timestampsToSearch.entrySet()) {
             SortedSet<OffsetAndTimestamp> offsets =
-                offsetForTimes.getOrDefault(entry.getKey(), new TreeSet<>());
```

Comment:
This is the main change, `new TreeSet<>()` doesn't have a comparator so the order can be messed up here.